src/components/CheckboxWithLabel.js

```diff
@@ -23,26 +27,46 @@ const propTypes = {
     /** Component to display for label */
     LabelComponent: PropTypes.func,
 
-    /** Should the input be styled for errors  */
-    hasError: PropTypes.bool,
 
     /** Error text to display */
     errorText: PropTypes.string,
+
+    /** Indicates that the input is being used with the Form component */
+    isFormInput: PropTypes.bool,
+
+    /**
+     * The ID used to uniquely identify the input
+     *
+     * @param {Object} props - props passed to the input
+     * @returns {Object} - returns an Error object if isFormInput is supplied but inputID is falsey or not a string
+     */
+    inputID: props => FormUtils.getInputIDPropTypes(props),
+
+    /** Saves a draft of the input value when used in a form */
+    shouldSaveDraft: PropTypes.bool,
 };
 
 const defaultProps = {
+    onChange: () => {},
+    isFormInput: false,
+    inputID: undefined,
     style: [],
     label: undefined,
     LabelComponent: undefined,
-    hasError: false,
     errorText: '',
+    shouldSaveDraft: false,
 };
 
-const CheckboxWithLabel = (props) => {
+const CheckboxWithLabel = React.forwardRef((props, ref) => {
     const LabelComponent = props.LabelComponent;
     const defaultStyles = [styles.flexRow, styles.alignItemsCenter];
     const wrapperStyles = _.isArray(props.style) ? [...defaultStyles, ...props.style] : [...defaultStyles, props.style];
 
+    const onCheckboxPress = () => {
```

Comment:
https://github.com/Expensify/App/blob/main/STYLE.md#functions-1
> Any function declared in a **_library module_** should use the function myFunction keyword rather than const myFunction.

@tgolen this style rule doesn't apply here right?

Reply:
It doesn't apply if you get really strict about "library module", however, I think that the general principle should apply. I think for consistency, using `function` would be better. That style guide doesn't really explain why it was only mentioned for "library modules".

Something related to this is the name of the function. Functions should never be named for what they handle, only for what they do.

Reply:
bump @thesahindia

Reply:
Thanks for the bump. Can you suggest what name should I use for this function?

Reply:
@thesahindia `toggleCheckbox` sounds good